source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst

```diff
@@ -1109,21 +1109,47 @@ updateRSFromPrimary
         return
 
     # Election ids are ObjectIds, see
-    # "using setVersion and electionId to detect stale primaries"
+    # "Using electionId and setVersion to detect stale primaries"
     # for comparison rules.
 
-    # Null values for both electionId and setVersion are always considered less than
-    if serverDescription.electionId > topologyDescription.maxElectionId or (
-        serverDescription.electionId == topologyDescription.maxElectionId
-        and serverDescription.setVersion >= topologyDescription.maxSetVersion
-    ):
-        topologyDescription.maxElectionId = serverDescription.electionId
-        topologyDescription.maxSetVersion = serverDescription.setVersion
+    if serverDescription.maxWireVersion >= 17:
+      # Null values for both electionId and setVersion are always considered less than
+      if serverDescription.electionId > topologyDescription.maxElectionId or (
+          serverDescription.electionId == topologyDescription.maxElectionId
+          and serverDescription.setVersion >= topologyDescription.maxSetVersion
+      ):
+          topologyDescription.maxElectionId = serverDescription.electionId
+          topologyDescription.maxSetVersion = serverDescription.setVersion
+      else:
+          # Stale primary.
+          # replace serverDescription with a default ServerDescription of type "Unknown"
+          checkIfHasPrimary()
+          return
     else:
-        # Stale primary.
-        # replace serverDescription with a default ServerDescription of type "Unknown"
-        checkIfHasPrimary()
-        return
+      # Maintain old comparison rules, namely setVersion is checked before electionId
+      # see "Using electionId and setVersion to detect stale primaries"
+      # for comparison rules.
+
+      if topologyDescription.maxSetVersion is not null \
```

Comment:
Can you use () instead of \ for line continuation. Using backslash in Python is frowned upon and the old version of the spec didn't use it: https://github.com/mongodb/specifications/blob/6464c0692bd0f81d92d58da0f72bf1642a703056/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst

Reply:
Okay I think I got all the formatting right, the old code wasn't valid python, now it is so hopefully all the indentations and such are right